src/sentry/flags/providers.py

```diff
@@ -127,6 +148,30 @@ def handle_launchdarkly_event(
     ]
 
 
+def validate_launchdarkly_event(
+    request_data: bytes,
+    request_headers: HttpHeaders,
+    organization_id: int,
+) -> bool:
+    """Return "true" if the launchdarkly payload is valid."""
+    signature = request_headers.get("X-LD-Signature")
+    if signature is None:
+        return False
+
+    models = FlagWebHookSigningSecretModel.objects.filter(
+        organization_id=organization_id,
+        provider="launchdarkly",
+    ).all()
+    for model in models:
+        if hmac_sha256_hex_digest(model.secret, request_data) == signature:
+            return True
```

Comment:
This check is provider-specific right? Is it expected/common to have multiple secrets?

Reply:
Its possible but not common.

Reply:
Will we support deleting secrets in the future?